clients/src/main/resources/common/message/ListGroupsRequest.json

```diff
@@ -20,8 +20,14 @@
   // Version 1 and 2 are the same as version 0.
   //
   // Version 3 is the first flexible version.
-  "validVersions": "0-3",
+  //
+  // Version 4 adds the StatesFilter field (KIP-518).
+  "validVersions": "0-4",
   "flexibleVersions": "3+",
   "fields": [
+    { "name": "StatesFilter", "type": "[]string", "versions": "4+", "nullableVersions": "4+", "default": "null",
+      "about": "The states of the groups we want to list. If empty or null, all groups are returned with their state.", "fields": [
+      { "name": "Name", "type": "string", "versions": "4+", "about": "The name of the group state" }
```

Comment:
Seems this is ignored because the type of `StatesFilter` is `[]string`. Sort of feels like a bug in the schema validation that this gets ignored. In any case, should we remove it?